src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1273,27 +1273,50 @@ public bool Equals(ModuleHandle handle)
         private static void ValidateModulePointer(RuntimeModule module)
         {
             // Make sure we have a valid Module to resolve against.
-            if (module == null)
-                throw new InvalidOperationException(SR.InvalidOperation_NullModuleHandle);
+            if (module is null)
+            {
+                // Local function to allow inlining of simple null check.
+                ThrowInvalidOperationException();
+            }
+
+            [StackTraceHidden]
+            [DoesNotReturn]
+            static void ThrowInvalidOperationException() => throw new InvalidOperationException(SR.InvalidOperation_NullModuleHandle);
         }
 
         // SQL-CLR LKG9 Compiler dependency
         public RuntimeTypeHandle GetRuntimeTypeHandleFromMetadataToken(int typeToken) { return ResolveTypeHandle(typeToken); }
         public RuntimeTypeHandle ResolveTypeHandle(int typeToken)
         {
-            return new RuntimeTypeHandle(ResolveTypeHandleInternal(GetRuntimeModule(), typeToken, null, null));
+            return new RuntimeTypeHandle(ResolveTypeHandleInternal(GetRuntimeModule(), typeToken));
         }
         public RuntimeTypeHandle ResolveTypeHandle(int typeToken, RuntimeTypeHandle[]? typeInstantiationContext, RuntimeTypeHandle[]? methodInstantiationContext)
         {
             return new RuntimeTypeHandle(ModuleHandle.ResolveTypeHandleInternal(GetRuntimeModule(), typeToken, typeInstantiationContext, methodInstantiationContext));
         }
 
+        internal static RuntimeType ResolveTypeHandleInternal(RuntimeModule module, int typeToken)
+        {
+            try
+            {
+                RuntimeType? type = null;
+                ResolveType(new QCallModule(ref module), typeToken, null, 0, null, 0, ObjectHandleOnStack.Create(ref type));
+                return type!;
+            }
+            catch (Exception)
+            {
+                ValidateModulePointer(module);
```

Comment:
ValidateModulePointer cannot be moved here. The QCall assumes that the module is non-null.

Reply:
Moved up